engines/cryomni3d/versailles/logic.cpp

```diff
@@ -0,0 +1,1036 @@
+				_nextPlaceId = _currentPlaceId;
+			}
+			fimg->_exit = true;
+			break;
+		}
+	}
+}
+
+IMG_CB(41802) {
+	// Dispatch to the correct state
+	if (_gameVariables[GameVariables::kInkSpilled] &&
+	        !_gameVariables[GameVariables::kCollectedPaperOnTable]) {
+		// Draw paper with ink on it
+		ZonFixedImage::CallbackFunctor *functor =
+		    new Common::Functor1Mem<ZonFixedImage *, void, CryOmni3DEngine_Versailles>(this,
+		            &CryOmni3DEngine_Versailles::img_41802b);
+		fimg->changeCallback(functor);
+		return;
+	}
+	if (!_gameVariables[GameVariables::kInkSpilled] &&
+	        _gameVariables[GameVariables::kCollectedPaperOnTable]) {
+		// Draw table with ink in inkpot and without paper
+		ZonFixedImage::CallbackFunctor *functor =
+		    new Common::Functor1Mem<ZonFixedImage *, void, CryOmni3DEngine_Versailles>(this,
+		            &CryOmni3DEngine_Versailles::img_41802c);
+		fimg->changeCallback(functor);
+		return;
+	}
+	if (_gameVariables[GameVariables::kInkSpilled] &&
+	        _gameVariables[GameVariables::kCollectedPaperOnTable]) {
+		// Draw table with ink directly on table
+		ZonFixedImage::CallbackFunctor *functor =
+		    new Common::Functor1Mem<ZonFixedImage *, void, CryOmni3DEngine_Versailles>(this,
+		            &CryOmni3DEngine_Versailles::img_41802d);
+		fimg->changeCallback(functor);
+		return;
+	}
+
+	// There we have paper on table and ink is in its inkpot
+	fimg->load("12E2_20.GIF");
+	while (1) {
+		fimg->manage();
+		if (fimg->_exit || fimg->_zoneLow) {
+			fimg->_exit = true;
+			break;
+		}
+		if (fimg->_zoneUse && fimg->_currentZone == 1) {
+			// Collected paper
+			collectObject(95, fimg);
+			_gameVariables[GameVariables::kCollectedPaperOnTable] = 1;
+			setPlaceState(8, 1);
+			// Draw table with ink in inkpot and without paper
+			ZonFixedImage::CallbackFunctor *functor =
+			    new Common::Functor1Mem<ZonFixedImage *, void, CryOmni3DEngine_Versailles>(this,
+			            &CryOmni3DEngine_Versailles::img_41802c);
+			fimg->changeCallback(functor);
+			break;
+		}
+		if (fimg->_zoneUse && fimg->_currentZone == 2) {
+			_gameVariables[GameVariables::kInkSpilled] = 1;
+			setPlaceState(8, 3);
+			// Draw paper with ink on it
+			ZonFixedImage::CallbackFunctor *functor =
+			    new Common::Functor1Mem<ZonFixedImage *, void, CryOmni3DEngine_Versailles>(this,
+			            &CryOmni3DEngine_Versailles::img_41802b);
+			fimg->changeCallback(functor);
+			break;
+		}
+		if (fimg->_usedObject && fimg->_currentZone == 0) {
+			unsigned int objID = fimg->_usedObject->idOBJ();
+			if (objID == 100) {
+				playInGameVideo("12E2_24");
+				// Force reload of the place
+				if (_nextPlaceId == -1u) {
+					_nextPlaceId = _currentPlaceId;
+				}
+				_inventory.removeByNameId(100);
+				// Revealed paper
+				collectObject(98, fimg);
+				_gameVariables[GameVariables::kGotRevealedPaper] = 1;
+				setGameTime(3, 1);
+			} else if (objID == 96) {
+				// Pamphlet about arts
+				playInGameVideo("PAP_BRUL");
```

Comment:
In the mac version this is actually called `PAP-BRUL` (with a dash and not an underscore), so it currently fails with 'WARNING: Failed to open movie file PAP_BRUL/PAP_BRUL.hns!` when getting to this point.
Is this a difference between the Mac and Windows/DOS version, or a typo here (and in the other places where this video is played)?

Reply:
It seems they mixed dashes and underscores during development. In PC version all paths are converted from dashes to underscore and there is no file with dash on the CDs.
I will have to do some substitution at various places...